test/tests/functional/pbs_soft_walltime.py

```diff
@@ -0,0 +1,634 @@
+            if 'Resource_List.soft_walltime' in RN_stat[0]:
+                wt = cvt_duration(RN_stat[0]['Resource_List.soft_walltime'])
+            elif 'Resource_List.walltime' in RN_stat[0]:
+                wt = cvt_duration(RN_stat[0]['Resource_List.walltime'])
+        else:
+            est = int(RN_stat[0][ATTR_stime])
+            if 'Resource_List.soft_walltime' in RN_stat[0]:
+                wt = int(RN_stat[0]['Resource_List.soft_walltime'])
+            elif 'Resource_List.walltime' in RN_stat[0]:
+                wt = int(RN_stat[0]['Resource_List.walltime'])
+
+        for j in jobs:
+            Jstat = self.server.status(JOB, id=j, attrib=a2)
+            wt2 = 0
+            if self.server.get_op_mode() == PTL_CLI:
+                strp = time.strptime(Jstat[0]['estimated.start_time'], '%c')
+                est2 = int(time.mktime(strp))
+                if 'Resource_List.soft_walltime' in Jstat[0]:
+                    wt2 = cvt_duration(Jstat[0]['Resource_List.soft_walltime'])
+                elif 'Resource_List.walltime' in Jstat[0]:
+                    wt2 = cvt_duration(Jstat[0]['Resource_List.walltime'])
+            else:
+                est2 = int(Jstat[0]['estimated.start_time'])
+                if 'Resource_List.soft_walltime' in Jstat[0]:
+                    wt2 = int(Jstat[0]['Resource_List.soft_walltime'])
+                elif 'Resource_List.walltime' in Jstat[0]:
+                    wt2 = int(Jstat[0]['Resource_List.walltime'])
+
+            self.assertEquals(est + wt, est2)
+            est = est2
+            wt = wt2
+
+    def setup_holidays(self, prime_offset, nonprime_offset):
+        """
+        Set up the holidays file for test execution.  This function will
+        first remove all entries in the holidays file and then add a year,
+        prime, and nonprime for all days.  The prime and nonprime entries
+        will be offsets from the current time.
+
+        This all is necessary because there are some holidays set by default.
+        The test should be able to be run on any day of the year.  If it is
+        run on one of these holidays, it will be nonprime time only.
+        """
+        # Delete all entries in the holidays file
+        self.scheduler.holidays_delete_entry('a')
+
+        lt = time.localtime(time.time())
+        self.scheduler.holidays_set_year(str(lt[0]))
+
+        now = int(time.time())
+        prime = time.strftime('%H%M', time.localtime(now + prime_offset))
+        nonprime = time.strftime('%H%M', time.localtime(now + nonprime_offset))
+
+        # set prime-time and nonprime-time for all days
+        self.scheduler.holidays_set_day('weekday', prime, nonprime)
+        self.scheduler.holidays_set_day('saturday', prime, nonprime)
+        self.scheduler.holidays_set_day('sunday', prime, nonprime)
+
+    def test_soft_walltime_perms(self):
+        """
+        Test to see if soft_walltime can't be submitted with a job or
+        altered by a normal user
```

Comment:
While I can do this, it is not necessary.  All jobs are submitted with user permissions.  If it can't be submitted as a normal user, it can't be submitted by one with higher permissions.  Do you still think it is necessary?

Reply:
Yes design explicitly talks about operator permissions so it should also be checked explicitly checked.

Reply:
@anamikau thank you for pointing this out.  It was an issue with my EDD.  I have updated the EDD to say that soft_walltime can not be requested at submit time.  With the update to the EDD, I don't think this test needs to be changed.  The fact that you can't request soft_walltime at submit time is because of existing behavior (all job submits use user permissions).

Reply:
Thanks for updating the EDD for job submission. I still think we need to extend the alterjob case below. I will include it in QA scenarios.